src/source/src/parser/mod.rs

```diff
@@ -306,16 +340,50 @@ impl SourceParserImpl {
     pub async fn parse(
         &self,
         payload: &[u8],
-        writer: SourceStreamChunkRowWriter<'_>,
+        builder: &mut SourceStreamChunkBuilder,
+        msg_timestamp: Option<i64>,
```

Comment:
I think there are some problems with this design, since it can't expand to other meta fields.